presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -690,7 +694,13 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
 
         checkTableIsWritable(table.get());
 
-        HiveStorageFormat hiveStorageFormat = extractHiveStorageFormat(table.get());
+        HiveStorageFormat hiveStorageFormat;
+        if (respectTableFormat) {
+            hiveStorageFormat = extractHiveStorageFormat(table.get());
+        }
+        else {
+            hiveStorageFormat = getHiveStorageFormat(ImmutableMap.of());
```

Comment:
This is going to return null and cause a NPE in HiveWritableTableHandle constructor